include/pybind11/numpy.h

```diff
@@ -243,6 +243,46 @@ template <typename T, size_t N> struct is_std_array<std::array<T, N>> : std::tru
 template <typename T> struct is_complex : std::false_type { };
 template <typename T> struct is_complex<std::complex<T>> : std::true_type { };
 
+template <typename T> struct array_info_scalar {
+    typedef T type;
+    static constexpr const bool is_array = false;
```

Comment:
Done for the code added in this pull request. There are other instances already in pybind11 that I haven't touched.